fendermint/vm/topdown/src/finality.rs

```diff
@@ -163,14 +180,27 @@ impl ParentFinalityProvider for InMemoryFinalityProvider {
             cache
         })?;
 
-        self.last_committed_finality.write(Some(finality.clone()))?;
+        self.last_committed_finality.write(Some(finality))?;
 
         Ok(())
     }
 }
 
 impl InMemoryFinalityProvider {
-    pub fn new(config: Config, committed_finality: Option<IPCParentFinality>) -> Self {
+    /// Creates an uninitialized provider
+    /// We need this because `fendermint` has yet to be initialized and might
+    /// not be able to provide an existing finality from the storage. This provider requires an
+    /// existing committed finality. Providing the finality will enable other functionalities.
+    pub fn uninitialized(config: Config) -> Self {
+        Self::new(config, None)
+    }
+
+    // creates an initialized provider
+    pub fn initialized(config: Config, committed_finality: IPCParentFinality) -> Self {
+        Self::new(config, Some(committed_finality))
+    }
```

Comment:
Okay so you have a conundrum because you wan to hand the finality provider to the application before it can be initialized, because it can at best get its data from genesis. 

I'd argue this is not the case. I'd say it's always possible as an operator to come up with a hash and height on the parent chain that we consider final, and add it to the `[ipc.topdown]` settings in the config file. And this doesn't have to be part of genesis, because if we start our node later, we can just use whatever is the latest checkpoint or final block from yesterday. 

There should be no reason to create an `uninitialized` provider.

Before the its' started, we can see if we already have in the ledger something newer than the config file, and use whichever is the higher height.

Reply:
I feel we should still keep the `uninitialized` and remove the `initialized`. First of all, I feel we should not use `config` because human errors are pretty common, like putting a value that is greater than the last committed finality. Also with the simplified error handling, if it's not initialized, it will just skip the current round of proposing.